### PR TITLE
ci: publish images to dockerhub

### PR DESCRIPTION
**Pull request by blumamir**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -9,6 +9,39 @@ permissions:
...
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and Push Docker Image for ${{ matrix.service }}
+        uses: docker/build-push-action@v5
+        with:
+          push: true
+          tags: keyval/odigos-${{ matrix.service }}:${{ env.TAG }}
```
Comment (rauno56) [inline .github/workflows/release.yml]: Maybe match the package name: 'odigos-demo-<service>'?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -9,6 +9,39 @@ permissions:
...
+        include:
+          - service: coupon
+          - service: frontend
+          - service: inventory
+          - service: membership
+          - service: pricing
+    runs-on: ubuntu-latest
+    steps:
+      - name: Store new version in env
+        run: echo "TAG=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
```
Comment (rauno56) [inline .github/workflows/release.yml:25]: Tagging with `${GITHUB_REF#refs/*/}` fails if branch name contains a `/` - I prefix my branches with `feat/` or `fix/`. Use `${GITHUB_REF##refs/*/}` to strip that as well.

**Review by rauno56 — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (blumamir) [reply .github/workflows/release.yml]: good catch

File: .github/workflows/release.yml (hunk 2 above)
Comment (blumamir) [reply .github/workflows/release.yml:25]: The workflow should run for version tags only: 

```
on:
  push:
    tags:
      - "v*"
```

So I think it should be ok, and this is also how it's used in all the other workflow jobs.

But good to know :)

**Pull request merged**